src/goto-programs/goto_inline_class.cpp

```diff
@@ -43,28 +43,21 @@ void goto_inlinet::parameter_assignments(
   // iterates over the operands
   exprt::operandst::const_iterator it1=arguments.begin();
 
-  const code_typet::parameterst &parameter_types=
-    code_type.parameters();
-
-  // iterates over the types of the parameters
-  for(const auto &parameter : parameter_types)
+  // iterates over the parameters
+  for(const auto &identifier : parameter_identifiers)
   {
-    // this is the type the n-th argument should be
-    const typet &par_type = parameter.type();
-
-    const irep_idt &identifier=parameter.get_identifier();
-
     DATA_INVARIANT(
       !identifier.empty(),
       source_location.as_string() + ": no identifier for function parameter");
 
-    {
-      const symbolt &symbol=ns.lookup(identifier);
+    const symbolt &symbol = ns.lookup(identifier);
 
-      goto_programt::targett decl = dest.add(
-        goto_programt::make_decl(symbol.symbol_expr(), source_location));
-      decl->code.add_source_location()=source_location;
-    }
+    // this is the type the n-th argument should have
+    const typet &par_type = symbol.type;
```

Comment:
:pick: `param_type` I think would be a lot clearer

Reply:
Will do (although this is just existing code moved around).

Reply:
I have added a second commit just taking care of the renaming.